integration/base/src/main/java/com/inrupt/client/integration/base/Utils.java

```diff
@@ -166,6 +173,96 @@ public static Set<Triple> publicAgentPolicyTriples(final URI acl) {
         return triples;
     }
 
+    public static void createPublicContainer(final SolidSyncClient authClient, final URI publicContainerURI) {
+        if (!exists(authClient, publicContainerURI)) {
+            try (SolidContainer newContainer = new SolidContainer(publicContainerURI)) {
+                try (final SolidContainer container = authClient.create(newContainer)) {
+                    container.getMetadata().getAcl().ifPresent(acl -> {
+                        int i = 0; // in case we cannot read the acl, we try 2 times before giving up
+                        while (i <= 1) {
+                            try (final SolidRDFSource acr = authClient.read(acl, SolidRDFSource.class)) {
+                                Utils.publicAgentPolicyTriples(acl)
+                                        .forEach(acr.getGraph()::add);
+                                authClient.update(acr);
+                                i = 2;
+                            } catch (SolidClientException ignored) {
+                                i++;
+                            }
+                        }
+                    });
+                }
+            }
+        }
+    }
+
+    public static boolean exists(final SolidSyncClient authClient, final URI containerURI) {
+        final var headReq = Request.newBuilder(containerURI)
+                .HEAD()
+                .build();
+        final var resCheckIfExists =
+                authClient.send(headReq, Response.BodyHandlers.discarding());
+        return Utils.isSuccessful(resCheckIfExists.statusCode());
+    }
+
+    public static void createContainer(final SolidSyncClient authClient, final URI containerURI) {
+        if (!exists(authClient, containerURI)) {
+            final var requestCreate = Request.newBuilder(containerURI)
+                    .header(Utils.CONTENT_TYPE, Utils.TEXT_TURTLE)
+                    .header("Link", Headers.Link.of(LDP.RDFSource, "type").toString())
+                    .PUT(Request.BodyPublishers.noBody())
+                    .build();
+            final var resCreate =
+                    authClient.send(requestCreate, Response.BodyHandlers.discarding());
+            assertTrue(Utils.isSuccessful(resCreate.statusCode()));
+        }
+    }
+
+    static void deleteContentsRecursively(final SolidSyncClient client, final URI url) {
+        if (exists(client, url)) {
+            deleteRecursive(client, url, new AtomicInteger(0));
+        }
+    }
+
+    static void deleteRecursive(final SolidSyncClient client, final URI url, final AtomicInteger depth) {
```

Comment:
I would avoid the use of a shared AtomicInteger here; instead, you can just have a `final int depth` and perform arithmetic operations on that in the recursive calls

Reply:
Yes, the AtomicInteger was useful in an asynchronous context

Reply:
However, I cannot use it as `final int depth` but I do need a final in the lambda later. I did not find a better solution.